apps/nestjs-api/src/auth-entra-id/entra-id-login.middleware.ts

```diff
@@ -0,0 +1,85 @@
+import { AuthorizationUrlRequest, ResponseMode } from '@azure/msal-node'
+import { Injectable, NestMiddleware } from '@nestjs/common'
+import { NextFunction, Request, Response } from 'express'
+import { EntraIdConfigProvider } from './entra-id-config.provider'
+import { EntraIdService } from './entra-id.service'
+import { VerificationData } from './verification-data.interface'
+
+@Injectable()
+export class EntraIdLoginMiddleware implements NestMiddleware {
```

Comment:
Nice work here!

I've never used NestJS middleware. Is there a reason we're using it instead of a simple controller?

Reply:
It seemed like the easiest way to protect this route and make it reusable for other routes.